composer.json

```diff
@@ -61,12 +61,13 @@
         "symfony/expression-language": "~2.7|^3.0|^4.0",
         "symfony/css-selector": "^2.7|^3.0|^4.0",
         "phpoption/phpoption": "^1.1",
-        "jms/serializer-bundle": "^1.2|^2.0",
+        "jms/serializer-bundle": "^2.0|^3.0",
```

Comment:
What if someone has JMSSerializerBundle 1 installed? Would it make sense to add a conflict rule or detect this when the container is built?

Reply:
This is a interesting question, and I had this problem multiple times with optional dependencies.

Generally yes, sounds a good idea! Any idea of possible side effects?

Reply:
A conflict rule would mean that people using JMSSerializerBundle 1, but not its integration into FOSRestBundle will not be able to update to the next release of FOSRestBundle. I am not sure we really want to prevent that.

To work around that we could instead detect the version inside our DI extension class and throw an exception. But that would mean people would have already updated their dependencies before noticing the incompatibility.

Is version 1 of JMSSerializerBundle still maintained or is it already recommended to update?

Reply:
> Is version 1 of JMSSerializerBundle still maintained or is it already recommended to update?
JMSSerializerBundle 1.x is not maintained anymore

Reply:
> A conflict rule would mean that people using JMSSerializerBundle 1, but not its integration into FOSRestBundle will not be able to update to the next release of FOSRestBundle. I am not sure we really want to prevent that.

somebody is using jms + fos, but without integration....sounds weird to me. Probably a conflict rule is ok.

Reply:
> JMSSerializerBundle 1.x is not maintained anymore

Then the `conflict` rule looks like the best solution to me.

Reply:
I have already pushed the changes